src/services/avatars.ts

```diff
@@ -0,0 +1,166 @@
+import { Service } from "../service";
+
+interface assoc {
```

Comment:
To be more generic you can create "dictionary" type
something like this:
`type Dict<T> = { [key: string]: T } `